### PR TITLE
Generalizing hide associations

### PR DESCRIPTION
**Pull request by jreys**
#### Summary of Changes

@infograf768 , this is an improvement to your patch here:
https://github.com/joomla/joomla-cms/commit/e30ee0d14f8f4230ef1e6ff6a191f8fdedb5cf47#diff-57b1cb851829b953306d161dc65f7aa8R12, it's beeing done in one place rather than in several :smiley: 
#### Testing Instructions

Try editing a new item or an item that has it's language set to "*" -> All

### CODE REVIEW

File: layouts/joomla/edit/associations.php [hunk 1]
```diff
@@ -10,4 +10,11 @@
 defined('JPATH_BASE') or die;
 
 // JLayout for standard handling of associations fields in the administrator items edit screens.
-echo $displayData->getForm()->renderFieldset('item_associations');
+if ($displayData->getForm()->getValue('id') != 0 && $displayData->getForm()->getValue('language') != "*")
```
Comment (andrepereiradasilva) [inline layouts/joomla/edit/associations.php:13]: why not use 

``` php
$displayData->item->id != 0 && $displayData->item->language != '*'
```

like it waS?

Comment (jreys) [inline layouts/joomla/edit/associations.php:13]: those fields are protected, it won't work that way, tried it

Comment (jreys) [inline layouts/joomla/edit/associations.php]: it's an OR though, we want that behaviour in either cases

**Comment by infograf768** [comment]
This works fine here.
Now, the fields are present although hidden.

**Comment by infograf768** [comment]
Please propose the patch ASAP towards joomla staging.

**Comment by infograf768** [comment]
@jreys

**Comment by andrepereiradasilva** [comment]
merged in latest joomla staging

**Pull request closed without merging**